misk/src/main/kotlin/misk/MiskServiceModule.kt

```diff
@@ -108,7 +108,27 @@ class MiskCommonServiceModule : KAbstractModule() {
 
     val builder = ServiceGraphBuilder()
 
+    // Avoid adding services to the builder twice, while we migrate the API.
+    val trackedKeys = mutableSetOf<Key<*>>()
+
+    // Support the new ServiceModule API.
+    for (entry in serviceEntries) {
+      val service = injector.getInstance(entry.key)
+      builder.addService(entry.key, service)
+      trackedKeys += entry.key
+    }
+    for (edge in dependencies) {
+      builder.addDependency(dependent = edge.dependent, dependsOn = edge.dependsOn)
+    }
+    for (edge in enhancements) {
+      builder.enhanceService(toBeEnhanced = edge.toBeEnhanced, enhancement = edge.enhancement)
+    }
+
     // Support the deprecated DependantService interface.
+    //
+    // Since this loop operates on the set *all* services bound to the injector, we might already
```

Comment:
It shouldn’t. It should only see services installed like this:

```
multibind<Service>() ...

```

Reply:
`install(ServiceModule<>())` creates such a binding 😦

Reply:
change it to not?